### PR TITLE
Freemabd1/rw 133 - loading the demographics criteria tree for the cohort builder.

### PR DESCRIPTION
**Pull request by freemabd1**

### CODE REVIEW

File: api/src/main/resources/workbench.yaml [hunk 1]
```diff
@@ -650,6 +650,10 @@ definitions:
       selectable:
         description: specifies if user can search with
         type: boolean
+      conceptId:
```
Comment (jmandel) [inline api/src/main/resources/workbench.yaml]: From the docs, it's unclear how `code` and `conceptId` differ. Could the descriptions make this clear?

File: api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java [hunk 2]
```diff
@@ -47,11 +47,12 @@
             final Criteria criteria = new Criteria();
             criteria.setId(row.get(rm.get("id")).getLongValue());
             criteria.setType(row.get(rm.get("type")).getStringValue());
-            criteria.setCode(row.get(rm.get("code")).getStringValue());
+            criteria.setCode(row.get(rm.get("code")).isNull() ? null : row.get(rm.get("code")).getStringValue());
```
Comment (johnstill) [inline api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java]: If `code` is nullable, we'll want to remove it from criteria/required in `workbench.yaml`.  Also, why _is_ it nullable?  I believe you that it is, I'm just curious.  Is this the kind of thing where the criteria has to have at least one of `code` or `conceptId`?

Comment (freemabd1) [reply api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java]: Demographics don't have codes. Only icd9, icd10 and cpt have codes, but tables are structured the same for the sake of making the sql consistent across criteria tables.

File: api/src/main/resources/workbench.yaml (hunk 1 above)
Comment (freemabd1) [reply api/src/main/resources/workbench.yaml]: I will update the descriptions to try a make this clearer.

File: api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java (hunk 2 above)
Comment (danqrodney) [reply api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java]: Why not just combine the criteria tables (with an extra column for ICD9 vs. something else)?

Comment (freemabd1) [reply api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java]: Well... some of the tables aren't exactly the same but very close. I think it's easier for them the be in separate table for ease of rebuilding the tables and calculating the counts.

Comment (danqrodney) [reply api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java]: OK. My guess is that if the table structure is very similar, and the way you query it is very similar, the code will be simpler and easier to maintain if you use a single table (with optional columns that are only populated for certain types of criteria.)

Rebuilding parts of the table shouldn't be that hard to pull off... the tools that would normally populate a whole table will instead delete and insert rows in a subset of it.

Anyway, something to think about when you're working on unified data / API for criteria with the data browser team.

Comment (danqrodney) [reply api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java]: (I wouldn't necessarily change anything now if we're planning on changing the representation / switching to Cloud SQL moving forward.)

**Pull request merged**